validate-peer-dependencies.js

```diff
@@ -0,0 +1,85 @@
+const fs = require('fs');
+
+let dependencyBundlesConfig = {};
+let dependencyExternalsConfig = {};
+let packageConfig = {};
+
+function readDependencyBundles() {
+    // read the package.json to determine current 'bundle' peer dependencies
+    fs.readFile('dependency-bundles.json', 'utf8', (err, data) => {
+        if (err) {
+            throw `Unable to read 'dependency.bundles.json'.  Error: ${err}`;
+        }
+
+        dependencyBundlesConfig = JSON.parse(data);
+        readDependencyExternals();
+    });
+};
+
+function readDependencyExternals() {
+    // read the package.json to determine current 'external' peer dependencies
+    fs.readFile('dependency-externals.json', 'utf8', (err, data) => {
+        if (err) {
+            throw `Unable to read 'dependency.externals.json'.  Error: ${err}`;
+        }
+
+        dependencyExternalsConfig = JSON.parse(data);
+        readPackage();
+    })
+};
+
+function readPackage() {
+    // read the package.json to determine current 'bundle' peer dependencies
+    fs.readFile('package.json', 'utf8', (err, data) => {
+        if (err) {
+            throw `Unable to read 'package.json'.  Error: ${err}`;
+        }
+
+        packageConfig = JSON.parse(data);
+        validateDependencies();
+    });
+};
+
+function validateDependencies() {
+    const dependencies = packageConfig.dependencies;
+    const dependenciesNames = Object.keys(dependencies);
+
+    const dependenciesNeedingConfiguration = [];
+
+    for (let i = 0; i < dependenciesNames.length; i++) {
+        const dependencyName = dependenciesNames[i];
+
+        // ignore definitely typed dependencies
+        if (dependencyName[0] === "@") {
```

Comment:
Suggestion: Should we be checking that it starts with `@types`? Couldn't other valid NPM packages start with `@`?

Reply:
Good point!  Yea, I believe all Definitely Typed dependencies are scoped up `@types` https://medium.com/@KevinBGreene/surviving-the-typescript-ecosystem-working-with-types-and-type-definitions-3539baf26627#eecf